bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2335,36 +2335,30 @@ private static boolean checkIsLikeArrayType(Object sourceValue, BArrayType targe
 
         ArrayValue source = (ArrayValue) sourceValue;
         Type targetTypeElementType = targetType.getElementType();
+
         if (source.getType().getTag() == TypeTags.ARRAY_TAG) {
```

Comment:
I think we should revisit this if block.

I think what we've tried to do is optimize the array case, so maybe we can do
1. if `checkIsType(sourceElementType, targetTypeElementType, new ArrayList<>())` is true, return true.
2. else if `isNumericType(sourceElementType) && allowNumericConversion` is true, check if the target has exactly one numeric type.